test/libevmasm/Optimiser.cpp

```diff
@@ -615,12 +615,17 @@ BOOST_AUTO_TEST_CASE(cse_keccak256_twice_same_content)
 		Instruction::MSTORE,
 		u256(0x20),
 		Instruction::SWAP1,
+		Instruction::DUP2,
+		Instruction::SWAP1,
 		Instruction::KECCAK256,
 		u256(12),
-		Instruction::DUP3,
-		Instruction::SWAP1,
+		Instruction::DUP4,
+		Instruction::DUP2,
 		Instruction::MSTORE,
-		Instruction::DUP1
+		Instruction::SWAP2,
+		Instruction::SWAP1,
+		Instruction::SWAP2,
+		Instruction::KECCAK256
```

Comment:
Does this optimize if you use the other alternative of also storing the length in the cache?